src/jit/lsra.cpp

```diff
@@ -6093,14 +6011,37 @@ bool LinearScan::isRegInUse(RegRecord* regRec, RefPosition* refPosition)
     Interval* assignedInterval = regRec->assignedInterval;
     if (assignedInterval != nullptr)
     {
+        if (!assignedInterval->isActive)
+        {
+// This can only happen if we have a recentRefPosition active at this location that hasn't yet been freed
+// (Or, in the case of ARM, the other half of a double is either active or has an active recentRefPosition).
+#ifdef _TARGET_ARM_
+            if (refPosition->getInterval()->registerType == TYP_DOUBLE)
+            {
+                if (!isRefPositionActive(assignedInterval->recentRefPosition, refPosition->nodeLocation))
+                {
+                    RegRecord* otherHalfRegRec = findAnotherHalfRegRec(regRec);
+                    assert(otherHalfRegRec->assignedInterval->isActive ||
+                           isRefPositionActive(otherHalfRegRec->assignedInterval->recentRefPosition,
+                                               refPosition->nodeLocation));
+                }
+            }
+            else
+#else
+            {
+                assert(isRefPositionActive(assignedInterval->recentRefPosition, refPosition->nodeLocation));
+            }
+#endif
```

Comment:
This `#ifdef` logic is very confusing. Can you instead use:

```
            // This can only happen if we have a recentRefPosition active at this location that hasn't yet been freed
            // (Or, in the case of ARM, the other half of a double is either active or has an active recentRefPosition).
            CLANG_FORMAT_COMMENT_ANCHOR;

#ifdef _TARGET_ARM_
            if (refPosition->getInterval()->registerType != TYP_DOUBLE)
            {
                return true;
            }

            if (!isRefPositionActive(assignedInterval->recentRefPosition, refPosition->nodeLocation))
            {
                RegRecord* otherHalfRegRec = findAnotherHalfRegRec(regRec);
                assert(otherHalfRegRec->assignedInterval->isActive ||
                       isRefPositionActive(otherHalfRegRec->assignedInterval->recentRefPosition,
                                           refPosition->nodeLocation));
            }
#else // !_TARGET_ARM_
            assert(isRefPositionActive(assignedInterval->recentRefPosition, refPosition->nodeLocation));
            return true;
#endif // !_TARGET_ARM_
```

Reply:
Actually, my version (and your replacement) is incorrectly skipping the assert in the non-double ARM case.